anchore_engine/analyzers/hints.py

```diff
@@ -0,0 +1,407 @@
+            list_value = [value]
+        return list_value
+
+    def to_dict(self) -> dict:
+        self.check_required_fields()
+        self.normalize()
+        return {
+            "type": self.type,
+            "name": self.name,
+            "version": self.version,
+            "origin": self.origin,
+        }
+
+
+class RPMHint(Hint):
+    def __init__(self, pkg):
+        super().__init__(pkg, "rpm")
+        self.license = ensure_str(pkg.get("license", ""))
+        self.arch = ensure_str(pkg.get("arch", "x86_64"))
+        self.release = ensure_str(pkg.get("release", ""))
+        self.source = ensure_str(pkg.get("source", ""))
+        self.size = ensure_str(str(pkg.get("size", "0")))
+
+    def resolve_rpm_fields(self):
+        from anchore_engine.util.rpm import split_rpm_filename
+
+        p_name, p_parsed_version, p_release, p_epoch, p_arch = split_rpm_filename(
+            "{}-{}.{}.rpm".format(self.name, self.version, self.arch)
+        )
+        if self.name == p_parsed_version:
+            raise HintsError(
+                "hints package version for hints package ({}) is not valid for RPM package type".format(
+                    self.name
+                )
+            )
+        return p_name, p_parsed_version, p_release, p_epoch, p_arch
+
+    def normalize(self):
+        if not self.release or not self.source:
+            (
+                p_name,
+                p_parsed_version,
+                p_release,
+                p_epoch,
+                p_arch,
+            ) = self.resolve_rpm_fields()
+
+            self.version = p_parsed_version
+
+            if p_epoch:
+                self.version = "{}:{}".format(p_epoch, p_parsed_version)
+
+            self.release = p_release
+
+            if p_arch:
+                self.arch = p_arch
+
+            if self.source:
+                self.source = "{}-{}.{}.rpm".format(self.source, self.version, "src")
+            else:
+                self.source = "{}-{}-{}.{}.rpm".format(
+                    self.name, self.version, self.release, "src"
+                )
+
+        if self.arch == "amd64":
+            self.arch = "x86_64"
+
+    def to_dict(self) -> dict:
+        result = super().to_dict()
+        result.update(
+            {
+                "license": self.license,
+                "arch": self.arch,
+                "release": self.release,
+                "sourcepkg": self.source,
+                "size": self.size,
+            }
+        )
+        return result
+
+
+class PythonHint(Hint):
+    def __init__(self, pkg):
+        super().__init__(pkg, ensure_str(pkg.get("type", "python")).lower())
+        self.license = ensure_str(pkg.get("license", ""))
+        self.files = pkg.get("files", [])
+        self.metadata = json.dumps(pkg.get("metadata", {}))
+        self.location = ensure_str(pkg.get("location", ""))
+
+    def validate_files(self):
+        for inp in [self.files]:
+            if type(inp) is not list:
```

Comment:
usually you wouldn't check the type, you would check if it is an instance - or perhaps there is a good reason to check a type here?
```suggestion
            if not isinstance(inp, list):
```

Reply:
it was from old bad code that I didn't want to break, can use isinstance